# Analysis for the process of 5 exercises                   CLUSTER (START)

In [1]:
import pandas as pd
import numpy as np
import seaborn as sn
import scipy.stats as st
import statsmodels.stats.multicomp as ml
from scipy.stats import f_oneway
from statsmodels.stats.multicomp import pairwise_tukeyhsd
import scikit_posthocs as sp
import matplotlib
import matplotlib.pyplot as plt 
import scipy.stats as stats
%matplotlib inline

In [2]:
#EΙΣΑΓΩΓΗ ΔΕΔΟΜΕΝΩΝ
df = pd.read_excel('MOOC3.xlsx', sheet_name= "GENERAL", skiprows=1)
df.head()
#ΕΠΙΛΟΓΗ ΣΥΓΚΕΚΡΙΜΕΝΩΝ ΣΤΗΛΩΝ (total_grade,Cluster (START),Cluster (AFTER),Subgroup,QUIZ  1,QUIZ  2,QUIZ  3,QUIZ  4,QUIZ  5)
df4=df[['total_grade', 'Cluster (START)', 'Cluster (AFTER)', 'Subgroup', 'Άσκηση  1','Άσκηση 2','Άσκηση 3','Άσκηση 4','Άσκηση5']]
#ΕΠΙΛΟΓΗ SUBGROUP ΝΑ ΕΙΝΑΙ ΙΣΣΟ ΜΕ 1 
#one= df.Subgroup==1
#df_by_one= df[one]
df4

total_grade  Cluster (START)  Cluster (AFTER)  Subgroup Άσκηση  1  \
0           2.71              2.0              NaN       1.0        10   
1            2.7              2.0              NaN       1.0         -   
2              -              NaN              NaN       NaN         -   
3              -              NaN              NaN       NaN         -   
4           9.91              2.0              1.0       1.0        10   
...          ...              ...              ...       ...       ...   
2655           -              NaN              NaN       NaN         -   
2656        1.23              2.0              NaN       1.0         -   
2657        8.73              2.0              2.0       1.0        10   
2658           -              NaN              NaN       NaN         -   
2659        1019              NaN              NaN       NaN       NaN   

     Άσκηση 2 Άσκηση 3 Άσκηση 4 Άσκηση5  
0           -        -        -       -  
1           -        -        -       -  
2           -        -        -       -  
3           -        -        -       -  
4          10       10       10      10  
...       ...      ...      ...     ...  
2655        -        -        -       -  
2656        -        -        -       -  
2657        8       10        -      10  
2658        -        -        -       -  
2659      NaN      NaN      NaN     NaN  

[2660 rows x 9 columns]

In [3]:
#EΛΕΓΧΟΣ ΓΙΑ ΚΑΘΕ ΜΑΘΗΤΗ ΑΝ ΕΧΕΙ ΑΠΑΝΤΗΣΕΙ ΣΕ ΤΟΥΛΑΧΙΣΤΟΝ 2 ΑΣΚΗΣΕΙΣ ΣΥΝΟΛΙΚΑ. ΑΝ ΕΧΕΙ <2 AΠΑΝΤΗΜΕΝΑ QUIZ ΤΟ ΚΑΝΩ DROP 

exercise_list=['Άσκηση  1','Άσκηση 2','Άσκηση 3','Άσκηση 4','Άσκηση5']


for i in range(df4.shape[0]):
    counter = 0
    if df4.at[i,'Subgroup'] == 1:
        
        for j in exercise_list:
            
            if df4.at[i,j]!="-":             
                counter+=1                       
        if counter<2:
            
            df4.drop(df.index[i], inplace=True)           
    else:
        df4.drop(df.index[i], inplace=True)
        
          
df4

C:\Users\stell\anaconda3\lib\site-packages\pandas\core\frame.py:4163: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().drop(


total_grade  Cluster (START)  Cluster (AFTER)  Subgroup Άσκηση  1  \
4           9.91              2.0              1.0       1.0        10   
10          9.07              3.0              3.0       1.0        10   
15           3.9              1.0              NaN       1.0        10   
19          8.67              3.0              1.0       1.0         8   
21          3.41              1.0              NaN       1.0        10   
...          ...              ...              ...       ...       ...   
2642        9.66              4.0              2.0       1.0        10   
2647         9.2              3.0              3.0       1.0        10   
2648        8.51              3.0              1.0       1.0         4   
2652        8.89              3.0              3.0       1.0        10   
2657        8.73              2.0              2.0       1.0        10   

     Άσκηση 2 Άσκηση 3 Άσκηση 4 Άσκηση5  
4          10       10       10      10  
10         10       10       10       5  
15          2        -        -       -  
19          8       10        3      10  
21          2        -        -       -  
...       ...      ...      ...     ...  
2642        8       10       10      10  
2647        6       10        9       8  
2648        2       10       10      10  
2652        8       10        -      10  
2657        8       10        -      10  

[867 rows x 9 columns]

In [4]:
df4ex=df4[['Άσκηση  1','Άσκηση 2','Άσκηση 3','Άσκηση 4','Άσκηση5']]
df4ex.replace(to_replace='-',value=float('nan'), inplace=True)
df4ex['mean'] = df4ex.mean(axis=1, skipna=True)

df4info=df4[['total_grade', 'Cluster (START)', 'Subgroup']]
#dfclu=df[df.columns[228:229:1]]

result= [df4info, df4ex]
dfex_total= pd.concat(result, axis=1)

dfex_total

C:\Users\stell\anaconda3\lib\site-packages\pandas\core\frame.py:4379: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().replace(
<ipython-input-4-d95e0323a649>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df4ex['mean'] = df4ex.mean(axis=1, skipna=True)


total_grade  Cluster (START)  Subgroup  Άσκηση  1  Άσκηση 2  Άσκηση 3  \
4           9.91              2.0       1.0       10.0      10.0      10.0   
10          9.07              3.0       1.0       10.0      10.0      10.0   
15           3.9              1.0       1.0       10.0       2.0       NaN   
19          8.67              3.0       1.0        8.0       8.0      10.0   
21          3.41              1.0       1.0       10.0       2.0       NaN   
...          ...              ...       ...        ...       ...       ...   
2642        9.66              4.0       1.0       10.0       8.0      10.0   
2647         9.2              3.0       1.0       10.0       6.0      10.0   
2648        8.51              3.0       1.0        4.0       2.0      10.0   
2652        8.89              3.0       1.0       10.0       8.0      10.0   
2657        8.73              2.0       1.0       10.0       8.0      10.0   

      Άσκηση 4  Άσκηση5  mean  
4         10.0     10.0  10.0  
10        10.0      5.0   9.0  
15         NaN      NaN   6.0  
19         3.0     10.0   7.8  
21         NaN      NaN   6.0  
...        ...      ...   ...  
2642      10.0     10.0   9.6  
2647       9.0      8.0   8.6  
2648      10.0     10.0   7.2  
2652       NaN     10.0   9.5  
2657       NaN     10.0   9.5  

[867 rows x 9 columns]

In [6]:
#Cluster Start 1 (LOW>4)
cs1ex=dfex_total.groupby('Cluster (START)').get_group(1)

cs1ex
col=cs1ex.columns
exlow=cs1ex[col[8:9:1]]
exlow.dropna(inplace=True)#κρατάμε μόνο την στήλη των μέσων όρων

meanexlow = exlow.mean(axis=0)#μεσος ορος βαθμολογίας μαθητων low στις 5 ασκήσεις

exlow_list=[]
for i, row in exlow.iterrows():
    exlow_list.append(exlow.at[i,'mean'])
exlow_list

<ipython-input-6-d6092f50945d>:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  exlow.dropna(inplace=True)#κρατάμε μόνο την στήλη των μέσων όρων


[6.0,
 6.0,
 5.6,
 8.666666666666666,
 7.6,
 9.6,
 9.8,
 8.4,
 9.25,
 9.0,
 2.8,
 10.0,
 4.4,
 8.4,
 7.0,
 7.4,
 5.6,
 8.8,
 7.2,
 10.0,
 8.5,
 6.0,
 5.8,
 8.4,
 9.8,
 9.666666666666666,
 4.6,
 8.0,
 10.0,
 10.0,
 9.2,
 9.4,
 9.6,
 8.4,
 8.8,
 9.4,
 9.4,
 7.6,
 10.0,
 8.6,
 7.5,
 7.4,
 8.8,
 10.0,
 9.8,
 8.0,
 2.8,
 8.2,
 7.5]

In [7]:
#Cluster Start 2 (MEDIUM 4-5)
cs2ex=dfex_total.groupby('Cluster (START)').get_group(2)

cs2ex
col=cs2ex.columns
exmedium=cs2ex[col[8:9:1]]
exmedium.dropna(inplace=True)#κρατάμε μόνο την στήλη των μέσων όρων

meanexmedium = exmedium.mean(axis=0)#μεσος ορος βαθμολογίας μαθητων medium στις 5 ασκήσεις

exmedium_list=[]
for i, row in exmedium.iterrows():
    exmedium_list.append(exmedium.at[i,'mean'])
exmedium_list

<ipython-input-7-cfc2fcdb56c4>:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  exmedium.dropna(inplace=True)#κρατάμε μόνο την στήλη των μέσων όρων


[10.0,
 9.2,
 6.8,
 9.6,
 10.0,
 9.5,
 8.8,
 7.6,
 8.6,
 10.0,
 9.0,
 7.0,
 9.6,
 9.0,
 10.0,
 8.4,
 10.0,
 10.0,
 10.0,
 10.0,
 10.0,
 2.5,
 10.0,
 10.0,
 10.0,
 9.8,
 9.6,
 8.0,
 10.0,
 10.0,
 9.4,
 5.8,
 6.6,
 10.0,
 8.4,
 10.0,
 9.2,
 9.6,
 9.6,
 8.8,
 8.8,
 10.0,
 8.6,
 9.6,
 10.0,
 9.0,
 6.2,
 10.0,
 4.4,
 8.2,
 9.6,
 8.0,
 9.4,
 10.0,
 5.6,
 8.0,
 9.2,
 7.2,
 9.5,
 8.6,
 8.8,
 9.4,
 8.25,
 8.666666666666666,
 9.6,
 10.0,
 10.0,
 9.6,
 2.5,
 6.0,
 9.2,
 8.8,
 9.333333333333334,
 8.8,
 10.0,
 9.6,
 8.2,
 9.6,
 10.0,
 10.0,
 8.0,
 9.4,
 9.0,
 10.0,
 6.0,
 9.0,
 8.75,
 3.0,
 6.0,
 9.4,
 9.6,
 9.6,
 8.8,
 8.8,
 8.5,
 8.0,
 10.0,
 9.2,
 9.8,
 8.0,
 9.2,
 8.0,
 9.4,
 9.8,
 8.8,
 8.4,
 6.0,
 8.0,
 4.0,
 6.0,
 9.2,
 8.4,
 8.4,
 9.8,
 10.0,
 8.4,
 10.0,
 8.0,
 8.6,
 8.0,
 8.6,
 6.75,
 9.0,
 9.6,
 9.0,
 10.0,
 8.666666666666666,
 8.2,
 10.0,
 9.6,
 9.6,
 8.6,
 9.2,
 10.0,
 8.6,
 2.4,
 10.0,
 6.0,
 9.6,
 8.8,
 9.0,
 6.0,
 8.4,
 10.0,
 6.2,
 8.2,
 8.2,
 10.0,
 8.8,
 9.2,
 7.4,
 9.2,
 7.0,
 7

In [8]:
#Cluster Start 3 (LARGE 5-6)
cs3ex=dfex_total.groupby('Cluster (START)').get_group(3)

cs3ex
col=cs3ex.columns
exlarge=cs3ex[col[8:9:1]]
exlarge.dropna(inplace=True)#κρατάμε μόνο την στήλη των μέσων όρων

meanexlarge = exlarge.mean(axis=0)#μεσος ορος βαθμολογίας μαθητων large στις 5 ασκήσεις

exlarge_list=[]
for i, row in exlarge.iterrows():
    exlarge_list.append(exlarge.at[i,'mean'])
exlarge_list

<ipython-input-8-62b10660a87e>:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  exlarge.dropna(inplace=True)#κρατάμε μόνο την στήλη των μέσων όρων


[9.0,
 7.8,
 9.4,
 8.6,
 5.0,
 7.8,
 7.8,
 9.2,
 7.6,
 10.0,
 9.8,
 10.0,
 8.6,
 6.2,
 10.0,
 6.2,
 8.2,
 9.5,
 9.0,
 9.6,
 9.6,
 9.6,
 8.6,
 10.0,
 7.6,
 7.0,
 9.5,
 9.6,
 10.0,
 9.2,
 5.6,
 6.0,
 6.6,
 9.6,
 8.4,
 9.0,
 7.4,
 7.2,
 7.2,
 9.6,
 5.4,
 10.0,
 9.5,
 8.8,
 8.4,
 3.4,
 8.6,
 8.6,
 5.8,
 10.0,
 8.6,
 9.4,
 9.6,
 7.6,
 10.0,
 8.4,
 6.333333333333333,
 7.4,
 6.8,
 9.4,
 9.8,
 8.4,
 9.2,
 8.6,
 10.0,
 9.0,
 9.2,
 10.0,
 9.4,
 7.0,
 10.0,
 8.8,
 9.0,
 10.0,
 3.0,
 10.0,
 7.8,
 10.0,
 7.0,
 6.666666666666667,
 9.2,
 9.8,
 10.0,
 9.6,
 10.0,
 10.0,
 8.8,
 10.0,
 9.4,
 10.0,
 5.6,
 10.0,
 7.6,
 7.6,
 3.5,
 8.2,
 7.4,
 7.6,
 7.6,
 8.4,
 9.0,
 8.5,
 10.0,
 6.4,
 5.8,
 8.0,
 9.25,
 7.0,
 9.8,
 9.0,
 7.2,
 10.0,
 8.6,
 5.5,
 10.0,
 9.4,
 8.0,
 9.6,
 9.4,
 7.0,
 8.0,
 6.0,
 10.0,
 9.8,
 9.2,
 6.4,
 8.6,
 7.6,
 6.4,
 9.8,
 9.4,
 8.5,
 10.0,
 8.25,
 10.0,
 4.0,
 9.2,
 8.6,
 8.8,
 8.0,
 7.333333333333333,
 6.0,
 9.4,
 9.2,
 9.4,
 7.6,
 9.4,
 10.0,
 8.6,
 8.6,
 8.8,
 10.0,
 10.0,
 9.8,
 9.

In [9]:
#Cluster Start 4 (XLARGE <6)
cs4ex=dfex_total.groupby('Cluster (START)').get_group(4)

cs4ex
col=cs4ex.columns
ex_xlarge=cs4ex[col[8:9:1]]
ex_xlarge.dropna(inplace=True)#κρατάμε μόνο την στήλη των μέσων όρων

meanex_xlarge = ex_xlarge.mean(axis=0)#μεσος ορος βαθμολογίας μαθητων xlarge στις 5 ασκήσεις

ex_xlarge_list=[]
for i, row in ex_xlarge.iterrows():
    ex_xlarge_list.append(ex_xlarge.at[i,'mean'])
ex_xlarge_list

<ipython-input-9-21f3c0643f41>:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  ex_xlarge.dropna(inplace=True)#κρατάμε μόνο την στήλη των μέσων όρων


[4.8,
 9.0,
 10.0,
 8.6,
 9.2,
 6.0,
 6.0,
 10.0,
 10.0,
 9.6,
 8.6,
 9.0,
 10.0,
 6.0,
 10.0,
 10.0,
 10.0,
 8.6,
 6.6,
 8.6,
 9.8,
 9.6,
 8.4,
 10.0,
 6.0,
 9.8,
 9.8,
 9.75,
 10.0,
 10.0,
 10.0,
 9.4,
 10.0,
 9.8,
 9.2,
 9.4,
 9.8,
 10.0,
 6.6,
 10.0,
 10.0,
 10.0,
 9.8,
 4.2,
 9.6]

# one way ANOVA test

In [10]:
# Shapiro-Wilk normality test
stats.shapiro(exlow_list), stats.shapiro(exmedium_list), stats.shapiro(exlarge_list), stats.shapiro(ex_xlarge_list)

#Όλες οι τιμές p είναι μικρότερες από το όριο a = 0,05, 
#επομένως δεν «αποτυγχάνουμε να απορρίψουμε» τη μηδενική υπόθεση 
#τα δείγματα προέρχονται από πληθυσμούς που δεν ακολουθούν την κανονική κατανομή

(ShapiroResult(statistic=0.8801467418670654, pvalue=0.0001312837703153491),
 ShapiroResult(statistic=0.8035686016082764, pvalue=2.183188130468863e-21),
 ShapiroResult(statistic=0.8657190203666687, pvalue=1.0612561822932834e-17),
 ShapiroResult(statistic=0.7066096067428589, pvalue=3.575032891944829e-08))

In [11]:
# Levene variance test  
stats.levene(exlow_list, exmedium_list, exlarge_list, ex_xlarge_list)

#Η τιμή p είναι μεγαλύτερη από το όριο a = 0,05, 
#επομένως «αποτυγχάνουμε να απορρίψουμε» τη μηδενική υπόθεση 
#τα δείγματα προέρχονται από πληθυσμούς με την ίδια διακύμανση

LeveneResult(statistic=1.3034166046625566, pvalue=0.2720549669279874)

In [12]:
#Apply one way ANOVA by calling the f_oneway() method
F, p = stats.f_oneway(exlow_list, exmedium_list, exlarge_list,ex_xlarge_list)
print('F statistic = {:5.3f} and probability p = {:5.3f}'.format(F, p))

F statistic = 3.610 and probability p = 0.013



# Kruskal Wallis test

In [13]:
stats.kruskal(exlow_list, exmedium_list, exlarge_list, ex_xlarge_list)

KruskalResult(statistic=14.009577642044023, pvalue=0.0028921449083786363)

# Dunn’s test

In [15]:
#perform Dunn's test using a Bonferonni correction for the p-values

data = [exlow_list, exmedium_list, exlarge_list, ex_xlarge_list]
data

sp.posthoc_dunn(data , p_adjust = 'bonferroni')

1         2         3         4
1  1.000000  0.034292  0.227419  0.003445
2  0.034292  1.000000  0.888843  0.381802
3  0.227419  0.888843  1.000000  0.071678
4  0.003445  0.381802  0.071678  1.000000

# Cohen's Effect Size

In [69]:
from numpy import mean
from numpy import var
from math import sqrt

n1,n2= len(exlow_list), len(exmedium_list)
s1,s2= var(exlow_list, ddof=1), var(exmedium_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(exlow_list), mean(exmedium_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

    

Medium Effect Size= 0.43706188565930665


In [70]:
n1,n2= len(exlow_list), len(exlarge_list)
s1, s2= var(exlow_list, ddof=1), var(exlarge_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(exlow_list), mean(exlarge_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)


Medium Effect Size= 0.3937223555778895


In [71]:
n1,n2= len(exlow_list), len(ex_xlarge_list)
s1,s2= var(exlow_list, ddof=1), var(ex_xlarge_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(exlow_list), mean(ex_xlarge_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

Large Effect Size= 0.5249371408554049


In [72]:
n1,n2= len(exmedium_list), len(exlarge_list)
s1,s2= var(exmedium_list, ddof=1), var(exlarge_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(exmedium_list), mean(exlarge_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

Very Small Effect Size= 0.06420406848771208


In [73]:
n1,n2= len(exmedium_list), len(ex_xlarge_list)
s1,s2= var(exmedium_list, ddof=1), var(ex_xlarge_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(exmedium_list), mean(ex_xlarge_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

Medium Effect Size= 0.15333174334698976


In [74]:
n1,n2= len(exlarge_list), len(ex_xlarge_list)
s1,s2= var(exlarge_list, ddof=1), var(ex_xlarge_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(exlarge_list), mean(ex_xlarge_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

Medium Effect Size= 0.2261805956063213


# In the first directions of the pairs the results of effect size are <0 , in the opposite directions of the pairs the results of effect size are >0 :
low-medium --> Very Small: -0.43706188565930665            medium-low --> Medium: 0.43706188565930665
low-large --> Very Small: -0.3937223555778895              large-low --> Medium: 0.3937223555778895
low-xlarge --> Very Small: -0.5249371408554049             xlarge-low --> Large: 0.5249371408554049
medium-large --> Very Small: 0.06420406848771208           large-medium --> Very Small: -0.06420406848771208
medium-xlarge --> Very Small: -0.15333174334698976         xlarge-medium --> Medium: 0.15333174334698976
large-xlarge --> Very Small: -0.2261805956063213           xlarge-large --> Medium: 0.2261805956063213